library/analytics/src/main/java/org/cru/godtools/analytics/model/LaunchAnalyticsActionEvent.kt

```diff
@@ -1,13 +1,16 @@
 package org.cru.godtools.analytics.model
 
-private const val ACTION_LAUNCH = "App Launch"
-private const val FIREBASE_EVENT_LAUNCH_PREFIX = "launch_"
+private const val ACTION_LAUNCH_PREFIX = "launch_"
+private const val ACTION_LAUNCH_AT_LEAST_PREFIX = "launch_atleast_"
 
-class LaunchAnalyticsActionEvent(private val launches: Int) : AnalyticsActionEvent(action = ACTION_LAUNCH) {
-    override fun isForSystem(system: AnalyticsSystem) = when (launches) {
-        1, 2, 3, 5, 10 -> system == AnalyticsSystem.FIREBASE
-        else -> false
-    }
+class LaunchAnalyticsActionEvent(private val launches: Int) : AnalyticsActionEvent(action = launches.buildAction()) {
+    override fun isForSystem(system: AnalyticsSystem) = launches > 0 && system == AnalyticsSystem.FIREBASE
+}
 
-    override val firebaseEventName get() = "$FIREBASE_EVENT_LAUNCH_PREFIX$launches"
+private fun Int.buildAction() = when (this) {
+    1, 2 -> "$ACTION_LAUNCH_PREFIX$this"
+    in 3..4 -> "${ACTION_LAUNCH_AT_LEAST_PREFIX}3"
+    in 5..9 -> "${ACTION_LAUNCH_AT_LEAST_PREFIX}5"
```

Comment:
I'm assuming we can just ignore these ```magic number``` warnings?

Reply:
yeah, I turned on some automated scanning, I'm still figuring out what is useful and then what I need to disable